sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtilTest.java

```diff
@@ -212,4 +215,60 @@ public void testInsertAll() throws Exception {
     // Each of the 25 rows has 1 byte for length and 30 bytes: '{"f":[{"v":"foo"},{"v":1234}]}'
     assertEquals("Incorrect byte count", 25L * 31L, totalBytes);
   }
+
+  static class ReadableCounter implements Counter {
+
+    private MetricName name;
+    private long value;
+
+    public ReadableCounter(MetricName name) {
+      this.name = name;
+      this.value = 0;
+    }
+
+    public long getValue() {
+      return value;
+    }
+
+    @Override
+    public void inc() {
+      ++value;
+    }
+
+    @Override
+    public void inc(long n) {
+      value += n;
+    }
+
+    @Override
+    public void dec() {
+      --value;
+    }
+
+    @Override
+    public void dec(long n) {
+      value -= n;
+    }
+
+    @Override
+    public MetricName getName() {
+      return name;
+    }
+  }
+
+  @Test
+  public void testNestedCounter() {
+    MetricName name1 = MetricName.named(this.getClass(), "metric1");
+    MetricName name2 = MetricName.named(this.getClass(), "metric2");
+    ReadableCounter counter1 = new ReadableCounter(name1);
+    ReadableCounter counter2 = new ReadableCounter(name2);
+    NestedCounter nested =
+        new NestedCounter(MetricName.named(this.getClass(), "nested"), counter1, counter2);
+    nested.inc();
+    nested.inc(10);
+    nested.dec();
+    nested.dec(2);
+    assertEquals(8, counter1.getValue());
+    assertEquals(8, counter2.getValue());
+  }
```

Comment:
Suggestion: 
increment one sub-counter before incrementing thru nested to check that both increments are recognized